### ISSUE TITLE
Prediction calibration step

### ISSUE THREAD

**Issue by kurtespinosa** [kyle29/Semantic-Textual-Similarity-Tensorflow#2]
Hi Kyle, I wonder if the prediction calibration step in the original paper (p. 4) has been implemented in your code? I don't seem to find it in your code although you did a conversion of the input values to lie between [0,1]. Particularly, this is their description of the calibration step:

Due to the simple construction of our similarity function, the predictions of our model are constrained to follow the exp(−x) curve and are thus not suited for these evaluation metrics. After training our model, we apply an additional nonparametric regression step to obtain better-calibrated predictions (with respect to MSE). Over the training set, the given labels (under original [1, 5] scale) are regressed against the univariate MaLSTM g-predicted relatedness as the sole covariate, and the fitted regression function is evaluated on the MaLSTM-predicted relatedness of the test pairs to produce adjusted final predictions. We use the classical local-linear estimator discussed in Fan and Gijbels (1992) with bandwidth selected using leave-one-out cross-validation. This calibration step serves as a minor correction for our restrictively simple similarity function (which is necessary to retain interpretability of the sentence representations).

Cheers,
Kurt

**Comment by kwangh** [comment]
Hi Kurt,

That was my issue. I could not manage to include the classical local-linear estimator discussed in Fan and Gijbels (1992). I used a simple linear multiplication to convert both the input values and the test values to lie between [0, 1].

Best,
Kyle

**Comment by kurtespinosa** [comment]
Yeah, I saw that. Thank you Kyle for that information.

Cheers,
Kurt
